public/app/features/alerting/NextGenAlertingPage.tsx

```diff
@@ -59,6 +60,10 @@ class NextGenAlertingPage extends PureComponent<Props> {
     }
   }
 
+  componentWillUnmount() {
+    this.props.cleanUpDefinitionState();
```

Comment:
Created a new Thunk that cleans up state when the component is unmounted. Calling it here.